projects/gateway2/translator/httproute/delegation_helpers.go

```diff
@@ -99,34 +109,44 @@ func filterDelegatedChildren(
 	return selected
 }
 
-func isDelegatedRouteMatch(
-	parent gwv1.HTTPRouteMatch,
+// isAllowedParent returns whether the parent specified by `parentRef` is allowed to delegate
+// to the child.
+//   - `childNs` is the namespace of the child route.
+//   - `childParentRefs` is the list of parent references on the child route. If this is empty, then
+//     there are no restrictions on which parents can delegate to this child. If it is not empty,
+//     then `parentRef` must be in this list in order for the parent to delegate to the child.
+func isAllowedParent(
 	parentRef types.NamespacedName,
-	child gwv1.HTTPRouteMatch,
 	childNs string,
-	parentRefs []gwv1.ParentReference,
+	childParentRefs []gwv1.ParentReference,
 ) bool {
```

Comment:
Would there be a need to check if parentRef is nil ?

Reply:
probably a good idea in general, but currently the place that calls this (in delegation.go) always initializes parentRef. since this func is not exported, i'm not too worried about it